### PR TITLE
enh(chore): package perl dependencies (cpan / net-curl / libssh-session ...)

### PR DESCRIPTION
**Pull request by hamzabessa**
## Description

package perl dependencies (cpan / net-curl / libssh-session ...)

**Fixes** MON-20179

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

### CODE REVIEW

File: .github/workflows/docker-builder-packaging-plugins.yml [hunk 1]
```diff
@@ -19,13 +19,32 @@ on:
...
+            project: plugins
+            distrib: alma9
+          - image: bullseye
+            runner: ubuntu-22.04
+            project: plugins
+            distrib: bullseye
+          - image: bullseye-arm64
+            runner: ["self-hosted", "collect-arm64"]
+            project: plugins
+            distrib: bullseye
```
Comment (kduret) [inline .github/workflows/docker-builder-packaging-plugins.yml:44]:
```suggestion
          - runner: ubuntu-22.04
            dockerfile: packaging-plugins-centos7
            image: packaging-plugins-centos7
          - runner: ubuntu-22.04
            dockerfile: packaging-plugins-alma8
            image: packaging-plugins-alma8
          - runner: ubuntu-22.04
            dockerfile: packaging-plugins-alma9
            image: packaging-plugins-alma9
          - runner: ubuntu-22.04
            dockerfile: packaging-plugins-bullseye
            image: packaging-plugins-bullseye
          - runner: ["self-hosted", "collect-arm64"]
            dockerfile: packaging-plugins-bullseye
            image: packaging-plugins-bullseye-arm64
```

File: .github/workflows/docker-builder-packaging-plugins.yml [hunk 2]
```diff
@@ -53,4 +72,4 @@ jobs:
           build-args: "REGISTRY_URL=${{ vars.DOCKER_PROXY_REGISTRY_URL }}"
           pull: true
           push: true
-          tags: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/packaging-${{ matrix.project }}-${{ matrix.distrib }}:latest
+          tags: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/packaging-${{ matrix.project }}-${{ matrix.image }}:latest
```
Comment (kduret) [inline .github/workflows/docker-builder-packaging-plugins.yml:75]:
```suggestion
          tags: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/${{ matrix.image }}:latest
```

Comment (kduret) [reply .github/workflows/docker-builder-packaging-plugins.yml:75]: update `file` input too

File: .github/workflows/perl-cpan-libraries.yml [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+        uses: actions/checkout@v4
+
+      - name: Delivery
+        uses: ./.github/actions/deb-delivery
+        with:
+          module_name: perl-cpan-libraries
+          distrib: ${{ matrix.distrib }}
+          artifactory_token: ${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}
+          cache_key: ${{ github.sha }}-${{ github.run_id }}-deb-${{ matrix.distrib }}
+          stability: ${{ needs.get-environment.outputs.stability }}
```
Comment (kduret) [inline .github/workflows/perl-cpan-libraries.yml:355]: where is the promote ???

File: .github/workflows/perl-crypt-argon2.yml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Promote ${{ matrix.distrib }} to stable
+        uses: ./.github/actions/promote-to-stable
+        with:
+          artifactory_token: ${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}
+          module: perl-cpan-libraries
+          distrib: ${{ matrix.distrib }}
+          stability: ${{ needs.get-environment.outputs.stability }}
```
Comment (kduret) [inline .github/workflows/perl-crypt-argon2.yml:197]: wrong promote
module name is not `perl-cpan-libraries`
missing promote for arm64
retrieve promote job from centreon/centreon

File: .github/workflows/perl-net-curl.yml [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Promote ${{ matrix.distrib }} to stable
+        uses: ./.github/actions/promote-to-stable
+        with:
+          artifactory_token: ${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}
+          module: perl-cpan-libraries
+          distrib: ${{ matrix.distrib }}
+          stability: ${{ needs.get-environment.outputs.stability }}
```
Comment (kduret) [inline .github/workflows/perl-net-curl.yml:195]: wrong promote
module name is not `perl-cpan-libraries`
missing promote for arm64
retrieve promote job from centreon/centreon

File: .github/workflows/perl-libssh-session.yml [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Promote ${{ matrix.distrib }} to stable
+        uses: ./.github/actions/promote-to-stable
+        with:
+          artifactory_token: ${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}
+          module: perl-cpan-libraries
+          distrib: ${{ matrix.distrib }}
+          stability: ${{ needs.get-environment.outputs.stability }}
```
Comment (kduret) [inline .github/workflows/perl-libssh-session.yml:195]: wrong promote
module name is not `perl-cpan-libraries`
missing promote for arm64
retrieve promote job from centreon/centreon

File: .github/workflows/plink.yml [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Promote ${{ matrix.distrib }} to stable
+        uses: ./.github/actions/promote-to-stable
+        with:
+          artifactory_token: ${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}
+          module: perl-cpan-libraries
+          distrib: ${{ matrix.distrib }}
+          stability: ${{ needs.get-environment.outputs.stability }}
```
Comment (kduret) [inline .github/workflows/plink.yml:149]: wrong promote
module name is not `perl-cpan-libraries`

File: .github/workflows/perl-keepass-reader.yml [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Promote ${{ matrix.distrib }} to stable
+        uses: ./.github/actions/promote-to-stable
+        with:
+          artifactory_token: ${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}
+          module: perl-cpan-libraries
+          distrib: ${{ matrix.distrib }}
+          stability: ${{ needs.get-environment.outputs.stability }}
```
Comment (kduret) [inline .github/workflows/perl-keepass-reader.yml:214]: wrong promote
module name is not `perl-cpan-libraries`

File: .github/workflows/perl-json-path.yml [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Promote ${{ matrix.distrib }} to stable
+        uses: ./.github/actions/promote-to-stable
+        with:
+          artifactory_token: ${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}
+          module: perl-cpan-libraries
+          distrib: ${{ matrix.distrib }}
+          stability: ${{ needs.get-environment.outputs.stability }}
```
Comment (kduret) [inline .github/workflows/perl-json-path.yml:184]: wrong promote
module name is not `perl-cpan-libraries`

File: .github/workflows/perl-filesys-smbclient.yml [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Promote ${{ matrix.distrib }} to stable
+        uses: ./.github/actions/promote-to-stable
+        with:
+          artifactory_token: ${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}
+          module: perl-cpan-libraries
+          distrib: ${{ matrix.distrib }}
+          stability: ${{ needs.get-environment.outputs.stability }}
```
Comment (kduret) [inline .github/workflows/perl-filesys-smbclient.yml:206]: wrong promote
module name is not `perl-cpan-libraries`

File: .github/workflows/perl-crypt-argon2.yml [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+    needs: [get-environment]
+    if: ${{ contains(fromJson('["stable"]'), needs.get-environment.outputs.stability) }}
+    runs-on: [self-hosted, common]
+    strategy:
+      matrix:
+        include:
+          - distrib: bullseye
+            arch: amd64
+          - distrib: bullseye
+            arch: arm64
```
Comment (kduret) [inline .github/workflows/perl-crypt-argon2.yml:189]: this was already missing in centreon repository
```suggestion
          - distrib: el8
            arch: amd64
          - distrib: el9
            arch: amd64
          - distrib: bullseye
            arch: amd64
          - distrib: bullseye
            arch: arm64
```

File: .github/workflows/perl-libssh-session.yml [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+    needs: [get-environment]
+    if: ${{ contains(fromJson('["stable"]'), needs.get-environment.outputs.stability) }}
+    runs-on: [self-hosted, common]
+    strategy:
+      matrix:
+        include:
+          - distrib: bullseye
+            arch: amd64
+          - distrib: bullseye
+            arch: arm64
```
Comment (kduret) [inline .github/workflows/perl-libssh-session.yml:187]: this was already missing in centreon repository
```suggestion
          - distrib: el8
            arch: amd64
          - distrib: el9
            arch: amd64
          - distrib: bullseye
            arch: amd64
          - distrib: bullseye
            arch: arm64
```

File: .github/actions/promote-to-stable/action.yml [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+      shell: bash
+
+    - name: Promote DEB package to stable
+      if: ${{ startsWith(inputs.distrib, 'bullseye') }}
+      run: |
+        echo "[DEBUG] - Distrib: ${{ inputs.distrib }}"
+
+        echo "[DEBUG] - Get path of testing DEB packages to promote to stable."
+        SRC_PATHS=$(jf rt s --include-dirs apt-plugins-stable-testing/pool/${{ inputs.module }}/*.deb | jq -r '.[].path')
```
Comment (kduret) [inline .github/actions/promote-to-stable/action.yml:64]:
```suggestion
        SRC_PATHS=$(jf rt s --include-dirs apt-plugins-testing/pool/${{ inputs.module }}/*.deb | jq -r '.[].path')
```

File: .github/actions/promote-to-stable/action.yml [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+          for SRC_PATH in ${SRC_PATHS[@]}; do
+            echo "[DEBUG] - Source path found: $SRC_PATH"
+          done
+        else
+          echo "[DEBUG] - No source path found."
+          continue
+        fi
+
+        echo "[DEBUG] - Build target path."
+        TARGET_PATH="apt-plugins-stable-${{ inputs.stability }}/pool/${{ inputs.module }}/"
```
Comment (kduret) [inline .github/actions/promote-to-stable/action.yml:76]:
```suggestion
        TARGET_PATH="apt-plugins-${{ inputs.stability }}/pool/${{ inputs.module }}/"
```

File: .github/workflows/perl-filesys-smbclient.yml [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+    name: sign rpm ${{ matrix.distrib }}
+    container:
+      image: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/rpm-signing:ubuntu
+      options: -t
+      credentials:
+        username: ${{ secrets.DOCKER_REGISTRY_ID }}
+        password: ${{ secrets.DOCKER_REGISTRY_PASSWD }}
+
+    steps:
+      - run: apt-get install -y zstd
```
Comment (kduret) [inline .github/workflows/perl-filesys-smbclient.yml:88]:
```suggestion
      - run: |
            apt-get update
            apt-get install -y zstd
```

File: .github/workflows/perl-keepass-reader.yml [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+    name: sign rpm ${{ matrix.distrib }}
+    container:
+      image: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/rpm-signing:ubuntu
+      options: -t
+      credentials:
+        username: ${{ secrets.DOCKER_REGISTRY_ID }}
+        password: ${{ secrets.DOCKER_REGISTRY_PASSWD }}
+
+    steps:
+      - run: apt-get install -y zstd
```
Comment (kduret) [inline .github/workflows/perl-keepass-reader.yml:84]:
```suggestion
      - run: |
            apt-get update
            apt-get install -y zstd
```

File: .github/workflows/plink.yml [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+    name: sign rpm ${{ matrix.distrib }}
+    container:
+      image: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/rpm-signing:ubuntu
+      options: -t
+      credentials:
+        username: ${{ secrets.DOCKER_REGISTRY_ID }}
+        password: ${{ secrets.DOCKER_REGISTRY_PASSWD }}
+
+    steps:
+      - run: apt-get install -y zstd
```
Comment (kduret) [inline .github/workflows/plink.yml:82]:
```suggestion
      - run: |
            apt-get update
            apt-get install -y zstd
```

**Review by kduret — APPROVED** [review]

**Review by bouda1 — APPROVED** [review]

**Review by lucie-dubrunfaut — APPROVED** [review]

**Pull request merged**